src/enzo/Grid_FDMCollapse.C

```diff
@@ -0,0 +1,145 @@
+/***********************************************************************
+/
+/  GRID CLASS (INITIALIZE THE GRID FOR A COLLAPSE TEST)
+/
+/  written by: Xinyu Li
+/  date:       May, 2019
+/  modified1:
+/
+/  PURPOSE: set up problem to collapse a FDM halo
+/
+/  RETURNS: FAIL or SUCCESS
+/
+************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ErrorExceptions.h"
+#include "macros_and_parameters.h"
+#include "typedefs.h"
+#include "global_data.h"
+#include "Fluxes.h"
+#include "GridList.h"
+#include "ExternalBoundary.h"
+#include "Grid.h"
+#include "CosmologyParameters.h"
+
+#define NTHETA 1000
+#define NR 1000
+
+/********************* PROTOTYPES *********************/
+int ReadFile(char *name, int Rank, int Dims[], int StartIndex[],
+       int EndIndex[], int BufferOffset[], float *buffer,
+       inits_type **tempbuffer, int Part, int Npart);
+ 
+int ReadIntFile(char *name, int Rank, int Dims[], int StartIndex[],
+    int EndIndex[], int BufferOffset[], int *buffer,
+    int **tempbuffer, int Part, int Npart);
+
+void ReadAttribute(hid_t dset_id, int *Attribute, char *AttributeName, FILE *log_fptr, int io_log);
+int ReadAttr(char *Fname, int *Rank, int Dims[], int *NSeg, int *LSeg, FILE *log_fptr);
+
+int GetUnits(float *DensityUnits, float *LengthUnits,
+	     float *TemperatureUnits, float *TimeUnits,
+	     float *VelocityUnits, FLOAT Time);
+int CosmologyComputeExpansionFactor(FLOAT time, FLOAT *a, FLOAT *dadt);
+
+/*******************************************************/
+#define READFILE ReadFile
+
+
+int grid::FDMCollapseInitializeGrid()
+{
+  /* declarations */
+
+  int dim, i, j, k, m, field, sphere, size, iden;
+  int RePsiNum, ImPsiNum, FDMDensNum;
+  float xdist,ydist,zdist;
+
+  /* create fields */
+
+  NumberOfBaryonFields = 0;
+  FieldType[NumberOfBaryonFields++] = Density;
+  FieldType[NumberOfBaryonFields++] = TotalEnergy;
+
+  int ivel = NumberOfBaryonFields;
+  FieldType[NumberOfBaryonFields++] = Velocity1;
+  if (GridRank > 1) 
+    FieldType[NumberOfBaryonFields++] = Velocity2;
+  if (GridRank > 2)
+    FieldType[NumberOfBaryonFields++] = Velocity3;
+
+  if (QuantumPressure){
+    FieldType[RePsiNum = NumberOfBaryonFields++] = RePsi;
+    FieldType[ImPsiNum = NumberOfBaryonFields++] = ImPsi;
+    FieldType[FDMDensNum = NumberOfBaryonFields++] = FDMDensity;
+  }
+  //printf("%d \n", NumberOfBaryonFields);
+
+  /* Set various units. */
+
+  const double Mpc = 3.0856e24, SolarMass = 1.989e33, GravConst = 6.67e-8,
+    pi = 3.14159, mh = 1.67e-24, kboltz = 1.381e-16, LightSpeed = 2.9979e10;
```

Comment:
Can these be replaced with the physical constants listed in `phys_constants.h`? (this centralization of all constants was a recent change to enzo-dev)

Reply:
removed, that part is never used in the code